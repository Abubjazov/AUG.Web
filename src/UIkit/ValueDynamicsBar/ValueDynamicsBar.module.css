.root {
  display: flex;
  flex-direction: column;
  width: 30%;
  min-width: 170px;
}

.wrapper {
  display: flex;
  gap: 5px;
  align-items: center;
  margin-top: 5px;
}

.red {
  color: var(--red);
}

.green {
  color: var(--green);
}

.title {
  font-size: 14px;
  font-weight: 400;
  line-height: 14px;
  color: var(--graylight);
  text-align: left;
}

@media (width <= 880px) {
  .root {
    width: calc(50% - 5%);
    min-width: unset;
  }
}
