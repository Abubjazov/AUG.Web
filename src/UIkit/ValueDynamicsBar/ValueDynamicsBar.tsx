import { FC } from 'react'

import SvgIcon from 'uikit/SvgIcon'
import { combineClasses as cc } from 'utils/combineClasses/combineClasses'
import { formatWithCurrency } from 'utils/formatWithCurrency/formatWithCurrency'

import styles from './ValueDynamicsBar.module.css'

export interface ValueDynamicsBarProps {
  userStyles?: string
  title?: string
  value: string
}

const ValueDynamicsBar: FC<ValueDynamicsBarProps> = ({
  userStyles = '',
  title,
  value,
}) => {
  const color = Number(value) >= 0 ? 'green' : 'red'

  return (
    <div className={styles.root}>
      {title && <span className={styles['title']}>{title}</span>}

      <div className={cc([styles.wrapper, userStyles, styles[color]])}>
        <SvgIcon icon={`${color}Triangle`} />

        {formatWithCurrency(value, '%')}
      </div>
    </div>
  )
}

export default ValueDynamicsBar
