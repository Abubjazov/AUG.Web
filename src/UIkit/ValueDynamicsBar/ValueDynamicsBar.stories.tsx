import type { Meta, StoryObj } from '@storybook/react'

import ValueDynamicsBar, { ValueDynamicsBarProps } from './ValueDynamicsBar'

import '/src/index.css'

const meta: Meta<ValueDynamicsBarProps> = {
  component: ValueDynamicsBar,
  title: 'UIkit/ValueDynamicsBar',
  parameters: {
    docs: {
      description: {
        component: 'Standard interface component - ValueDynamicsBar',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
    title: {
      description: 'Optional title text',
    },
    value: {
      description: 'Display value',
    },
  },
}

export default meta

type Story = StoryObj<ValueDynamicsBarProps>

export const Default: Story = {
  args: { title: 'Title text', value: '-13' },
  decorators: [Story => <Story />],
}
