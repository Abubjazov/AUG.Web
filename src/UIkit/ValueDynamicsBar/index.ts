import ValueDynamicsBar from './ValueDynamicsBar'

export default ValueDynamicsBar
