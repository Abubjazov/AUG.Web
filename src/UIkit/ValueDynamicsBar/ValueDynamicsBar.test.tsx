import { render } from '@testing-library/react'

import ValueDynamicsBar from './ValueDynamicsBar'

describe('ValueDynamicsBar', () => {
  test('should render ValueDynamicsBar default with positive value', () => {
    const { asFragment } = render(<ValueDynamicsBar value={'100'} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render ValueDynamicsBar default with negative value', () => {
    const { asFragment } = render(<ValueDynamicsBar value={'-100'} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render ValueDynamicsBar default title', () => {
    const { asFragment } = render(
      <ValueDynamicsBar value={'10.13'} title={'Title'} />,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
