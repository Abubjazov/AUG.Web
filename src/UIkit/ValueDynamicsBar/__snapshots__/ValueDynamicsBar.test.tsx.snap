// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ValueDynamicsBar > should render ValueDynamicsBar default title 1`] = `
<DocumentFragment>
  <div
    class="_root_951cff"
  >
    <span
      class="_title_951cff"
    >
      Title
    </span>
    <div
      class="_wrapper_951cff  _green_951cff"
    >
      <svg
        fill="#14C785"
        height="15px"
        viewBox="0 0 24 24"
        width="15px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.0566 16H5.94336C5.10459 16 4.68455 14.9782 5.27763 14.3806L11.3343 8.27783C11.7019 7.90739 12.2981 7.90739 12.6657 8.27783L18.7223 14.3806C19.3155 14.9782 18.8954 16 18.0566 16Z"
        />
      </svg>
      10.13 %
    </div>
  </div>
</DocumentFragment>
`;

exports[`ValueDynamicsBar > should render ValueDynamicsBar default with negative value 1`] = `
<DocumentFragment>
  <div
    class="_root_951cff"
  >
    <div
      class="_wrapper_951cff  _red_951cff"
    >
      <svg
        class="sc-aef7b723-0 eummbo"
        fill="#c50140"
        height="15px"
        viewBox="0 0 24 24"
        width="15px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.0566 8H5.94336C5.10459 8 4.68455 9.02183 5.27763 9.61943L11.3343 15.7222C11.7019 16.0926 12.2981 16.0926 12.6657 15.7222L18.7223 9.61943C19.3155 9.02183 18.8954 8 18.0566 8Z"
        />
      </svg>
      -100 %
    </div>
  </div>
</DocumentFragment>
`;

exports[`ValueDynamicsBar > should render ValueDynamicsBar default with positive value 1`] = `
<DocumentFragment>
  <div
    class="_root_951cff"
  >
    <div
      class="_wrapper_951cff  _green_951cff"
    >
      <svg
        fill="#14C785"
        height="15px"
        viewBox="0 0 24 24"
        width="15px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.0566 16H5.94336C5.10459 16 4.68455 14.9782 5.27763 14.3806L11.3343 8.27783C11.7019 7.90739 12.2981 7.90739 12.6657 8.27783L18.7223 14.3806C19.3155 14.9782 18.8954 16 18.0566 16Z"
        />
      </svg>
      100 %
    </div>
  </div>
</DocumentFragment>
`;
